pkg/detectors/sendgrid/sendgrid.go

```diff
@@ -36,53 +39,27 @@ func (s Scanner) Keywords() []string {
 func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
 	dataStr := string(data)
 
-	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
-
-	for _, match := range matches {
-		if len(match) != 2 {
-			continue
-		}
-		resMatch := strings.TrimSpace(match[1])
+	uniqueMatches := make(map[string]struct{})
+	for _, match := range keyPat.FindAllStringSubmatch(dataStr, -1) {
+		uniqueMatches[match[0]] = struct{}{}
+	}
```

Comment:
This is to avoid verifying the same matches found in same chunk multiple times right? 
@ahrav if this is helpful shouldn't we have this in all detectors then? Maybe another `[chore]`  task for you :smiling_face_with_tear:

Reply:
It's a bit of a bandaid solution to #2262. I mentioned this as well in https://github.com/trufflesecurity/trufflehog/pull/2812#discussion_r1594877538.

Reply:
Yea, we're aware of the task and are considering moving the caching logic into the engine to streamline processes by centralizing the logic, which would eliminate the need to replicate it across all detectors. For now, implementing the caching within each individual detector is an acceptable temporary solution.

Reply:
IMO it's probably another use case for SQLite, similar to #2696.